src/pages/api/checkAvail.tsx

```diff
@@ -5,35 +5,53 @@ import {
   makeRes,
```

Comment:
In general you should avoid making endpoints like this.  Try and follow REST standards. Looking into REST Resource naming conventions